src/Sulu/Bundle/ContentBundle/Document/Subscriber/PublishSubscriber.php

```diff
@@ -0,0 +1,227 @@
+        NodeHelperInterface $nodeHelper,
+        PropertyEncoder $propertyEncoder
+    ) {
+        $this->documentManager = $documentManager;
+        $this->liveSession = $liveSession;
+        $this->nodeHelper = $nodeHelper;
+        $this->propertyEncoder = $propertyEncoder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::PERSIST => ['createNodeInPublicWorkspace', -490],
+            Events::REMOVE => 'removeNodeFromPublicWorkspace',
+            Events::MOVE => 'moveNodeInPublicWorkspace',
+            Events::COPY => 'copyNodeInPublicWorkspace',
+            Events::REORDER => 'reorderNodeInPublicWorkspace',
+            Events::PUBLISH => ['setNodeFromPublicWorkspace', 512],
+            Events::FLUSH => 'flushPublicWorkspace',
+        ];
+    }
+
+    /**
+     * Creates the node with the same UUID in the public workspace if it does not exist yet. In case it does it will
+     * be renamed if necessary.
+     *
+     * @param PersistEvent $event
+     */
+    public function createNodeInPublicWorkspace(PersistEvent $event)
+    {
+        $node = $event->getNode();
+
+        if ($node->isNew()) {
+            $this->createNodesWithUuid($node);
+
+            return;
+        }
+
+        $liveNode = $this->getLiveNode($event->getDocument());
+        $nodeName = $node->getName();
+
+        if ($liveNode->getName() !== $nodeName) {
+            $liveNode->rename($nodeName);
+        }
+    }
+
+    /**
+     * Since deleting is not draftable the node will be deleted in the live session as soon as it is deleted in the
+     * default session.
+     *
+     * @param RemoveEvent $event
+     */
+    public function removeNodeFromPublicWorkspace(RemoveEvent $event)
+    {
+        $this->getLiveNode($event->getDocument())->remove();
+    }
+
+    /**
+     * Since moving is not draftable the node will also be moved in the live session immediately.
+     *
+     * @param MoveEvent $event
+     */
+    public function moveNodeInPublicWorkspace(MoveEvent $event)
+    {
+        $liveNode = $this->getLiveNode($event->getDocument());
+        $this->nodeHelper->move($liveNode, $event->getDestId(), $event->getDestName());
+    }
+
+    /**
+     * If a node is copied a node with the same UUID will be created in the live session.
+     *
+     * @param CopyEvent $event
+     */
+    public function copyNodeInPublicWorkspace(CopyEvent $event)
+    {
+        $this->createNodesWithUuid($event->getCopiedNode());
+    }
+
+    /**
+     * Reordering is also not draftable, and therefore also immediately applied to the live session.
+     *
+     * @param ReorderEvent $event
+     */
+    public function reorderNodeInPublicWorkspace(ReorderEvent $event)
+    {
+        $node = $event->getNode();
+
+        $this->nodeHelper->reorder($node, $event->getDestId());
+
+        // FIXME duplicating logic of OrderSubscriber, maybe move to NodeHelper?
```

Comment:
Was thinking about that... Why not implementing another `NodeHelper` solely for the `ContentBundle`? I don't see any real reason to combine all of these methods in one class anyway. Even if we merge the document manager repository later we still have to answer the question in which bundle this `NodeHelper` should go, and then we probably need more of these classes again...

Or we even think a lot smaller, and introduce separate classes for each operation: `MoveNodeHelper`, `CopyNodeHelper`, `ReorderNodeHelper` and so on. These classes should be easily assignable to a bundle, at the cost of having a lot more classes.